### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by braceal**
**Updates**
- Perplexity callback
- Make ThroughputMonitor wandb optional
- Templated HPC queue submission CLI (support for perlmutter and polaris
- Updated docker image to use hugging face v4.21.1 and install BLAST
- Explicitly include attention mask in forward function
- Architecture files for GPT-2, GPT-NeoX
- Update PyTorch lightning DeepSpeed API to DeepSpeedStrategy
- Setting drop_last=True for inference mode when generating embeddings

### CODE REVIEW

**Review by maxzvyagin — APPROVED** [review]
Looks good!
